### PR TITLE
Reintroduce default value for ignore_args argument of FixtureManager.getfixtureclosure

### LINKED ISSUE #11868: Latest 8.0.0 is breaking on 3.8+
### Comment by pytest maintainers: pytest-asyncio is not yet compatible with pytest 8; if you're using pytest-asyncio, stick with pytest<8 until the issue is resolved.

---

## Description

Prior to 8.0.0 all tests on 3.7 -- 3.11 were passing. Then today when 8.0.0 was released 3.8+ started failing.

See screen shot, rerunning tests that passed on 27 Jan, failed today on 28 Jan.
Screenshot from 2024-01-28 13-43-43

Failure is happening on 3.8+
Screenshot from 2024-01-28 13-43-25

The only requirements for the project are `pytest` and `pytest-asyncio`. For reference, the project is [FairAsyncRLock](https://github.com/Joshuaalbert/FairAsyncRLock). The git action workflow for the Ci/Cd is [here](https://github.com/Joshuaalbert/FairAsyncRLock/blob/main/.github/workflows/unittests.yml).

## Traceback

This is the problem. It's the same for all 3.8+.

```
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/_pytest/main.py", line 272, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>                          ^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/_pytest/main.py", line 325, in _main
INTERNALERROR>     config.hook.pytest_collection(session=session)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/pluggy/_hooks.py", line 501, in __call__
INTERNALERROR>     return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/pluggy/_manager.py", line 119, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/pluggy/_callers.py", line 138, in _multicall
INTERNALERROR>     raise exception.with_traceback(exception.__traceback__)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/pluggy/_callers.py", line 121, in _multicall
INTERNALERROR>     teardown.throw(exception)  # type: ignore[union-attr]
INTERNALERROR>     ^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/_pytest/logging.py", line 783, in pytest_collection
INTERNALERROR>     return (yield)
INTERNALERROR>             ^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/pluggy/_callers.py", line 121, in _multicall
INTERNALERROR>     teardown.throw(exception)  # type: ignore[union-attr]
INTERNALERROR>     ^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/_pytest/warnings.py", line 1[18](https://github.com/Joshuaalbert/FairAsyncRLock/actions/runs/7680017808/job/20940748975#step:6:19), in pytest_collection
INTERNALERROR>     return (yield)
INTERNALERROR>             ^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/pluggy/_callers.py", line 121, in _multicall
INTERNALERROR>     teardown.throw(exception)  # type: ignore[union-attr]
INTERNALERROR>     ^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/_pytest/config/__init__.py", line 1365, in pytest_collection
INTERNALERROR>     return (yield)
INTERNALERROR>             ^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/pluggy/_callers.py", line 102, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>           ^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/_pytest/main.py", line 336, in pytest_collection
INTERNALERROR>     session.perform_collect()
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/_pytest/main.py", line 799, in perform_collect
INTERNALERROR>     self.items.extend(self.genitems(node))
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/_pytest/main.py", line 942, in genitems
INTERNALERROR>     yield from self.genitems(subnode)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/_pytest/main.py", line 937, in genitems
INTERNALERROR>     rep, duplicate = self._collect_one_node(node, handle_dupes)
INTERNALERROR>                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/_pytest/main.py", line 825, in _collect_one_node
INTERNALERROR>     rep = collect_one_node(node)
INTERNALERROR>           ^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/_pytest/runner.py", line 566, in collect_one_node
INTERNALERROR>     ihook.pytest_collectstart(collector=collector)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/pluggy/_hooks.py", line 501, in __call__
INTERNALERROR>     return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/pluggy/_manager.py", line 1[19](https://github.com/Joshuaalbert/FairAsyncRLock/actions/runs/7680017808/job/20940748975#step:6:20), in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/pluggy/_callers.py", line 138, in _multicall
INTERNALERROR>     raise exception.with_traceback(exception.__traceback__)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/pluggy/_callers.py", line 102, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>           ^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/pytest_asyncio/plugin.py", line 6[26](https://github.com/Joshuaalbert/FairAsyncRLock/actions/runs/7680017808/job/20940748975#step:6:27), in pytest_collectstart
INTERNALERROR>     pyobject = collector.obj
INTERNALERROR>                ^^^^^^^^^^^^^
INTERNALERROR> AttributeError: 'Package' object has no attribute 'obj'
```

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #11868

### CODE REVIEW

**Comment by nicoddemus** [comment]
The failure is unrelated and fixed by #11889.

File: changelog/11868.bugfix.rst
Comment (bluetech) [inline changelog/11868.bugfix.rst:1]: It wasn't an accident really, so changing the wording a bit:

```suggestion
Reintroduced the default value for the ``ignore_args`` of internal ``FixtureManager.getfixtureclosure``: it was changed in 8.0.0 and this has broken a number of plugins.
```

**Review by bluetech — APPROVED** [review]

**Comment by YannickJadoul** [comment]
If I read this correctly, the order of arguments [also changed in #11416](https://github.com/pytest-dev/pytest/pull/11416/files#diff-49027cfd80e14edac9b0fae71f7228a408a09599f66a7815839ce8c3ae2ab84fR1530-R1535)?

Not my decision as to whether you want to make this backwards-compatible again with previous versions, but if you want to, the current PR likely isn't enough?

**Comment by bluetech** [comment]
@YannickJadoul you're right. I think the function changed too much (also in its semantics) that it's not really possible to maintain backwards compat for it. The situation with pytest-lazyfixtures is unfortunate since it seems it's abandoned without any way to transfer ownership. It probably needs a fork...

File: changelog/11868.bugfix.rst
Comment (nicoddemus) [reply changelog/11868.bugfix.rst:1]: I meant "non-intentional", meaning that we probably did not mean to drop the default argument and break external usages, figuring it was used internally only.

But I will apply your suggestion anyway, no need to over explain things.

**Comment by nicoddemus** [comment]
> @YannickJadoul you're right. I think the function changed too much (also in its semantics) that it's not really possible to maintain backwards compat for it. 

In that case, it is probably better to not merge this at all, as will only add to the confusion?

**Comment by bluetech** [comment]
Yes, I think we shouldn't merge it.

**Comment by YannickJadoul** [comment]
> The situation with pytest-lazyfixtures is unfortunate since it seems it's abandoned without any way to transfer ownership. It probably needs a fork...

Agreed, yes! That's why I explicitly stepped back from saying I'd like to see this change.
pytest-lazyfixtures will need more than just this PR, so I wouldn't consider it as a good reason to merge it.

**Comment by RonnyPfannschmidt** [comment]
should we consider integrating lazy-fixtures in core - its in line with the fixture configuration details i have in mind 

i already tried to reach out to the maintainer a while back and got no reply

**Comment by YannickJadoul** [comment]
> should we consider integrating lazy-fixtures in core - its in line with the fixture configuration details i have in mind
> 
> i already tried to reach out to the maintainer a while back and got no reply

I don't have massive amounts of time, but if this isn't super urgent and can take e.g. a few weeks, I would be interested in helping out here!
And maybe @youtux as well? (see TvoroG/pytest-lazy-fixture#63)

In the end, lazy-fixures is not a massive amount of code. I'd just need to look into all the previous issues and discussions had before, and get some input on whether the current implementation is still the best way forward or whether there's a more stable way of implementing it.

**Comment by RonnyPfannschmidt** [comment]
if pytest introduces a native object to statically request the request for another fixture as parameter (with or without a reconfigrue of the fixture parameters) implementation may look different

**Comment by nicoddemus** [comment]
@RonnyPfannschmidt @YannickJadoul there's a detailed topic already in #11532, I suggest to move the discussion over there.

**Pull request closed without merging**
